dotCMS/src/main/java/com/dotcms/rendering/velocity/util/VelocityUtil.java

```diff
@@ -0,0 +1,607 @@
+				context.put("visitor", visitor);
+
+			} catch (Exception nsue) {
+				Logger.error(VelocityUtil.class, nsue.getMessage(), nsue);
+			}
+		}
+		return context;
+
+	}
+
+	public static String mergeTemplate(String templatePath, Context ctx) throws ResourceNotFoundException, ParseErrorException, Exception{
+		VelocityEngine ve = VelocityUtil.getEngine();
+		Template template = null;
+		StringWriter sw = new StringWriter();
+		template = ve.getTemplate(templatePath);
+
+		template.merge(ctx, sw);
+
+		return sw.toString();
+		
+	}
+	
+	public static String eval(String velocity, Context ctx) throws ResourceNotFoundException, ParseErrorException, Exception{
+		VelocityEngine ve = VelocityUtil.getEngine();
+		StringWriter sw = new StringWriter();
+		ve.evaluate(ctx, sw, "velocity eval", velocity);
+		return sw.toString();
+		
+	}
+	private static ToolboxManager toolboxManager=null;
+    public static ToolboxManager getToolboxManager () {
+        if ( toolboxManager == null ) {
+            synchronized ( VelocityUtil.class ) {
+                if ( toolboxManager == null ) {
+                    toolboxManager = ServletToolboxManager.getInstance( Config.CONTEXT, Config.getStringProperty("TOOLBOX_MANAGER_PATH", "/WEB-INF/toolbox.xml"));
+                }
+            }
+
+        }
+        return toolboxManager;
+    }
+	
+	private static List<Language> languages =null;
+	
+	private static List<Language> getLanguages(){
+		if(languages ==null){
+			synchronized (VelocityUtil.class) {
+				if(languages ==null){
+					languages = APILocator.getLanguageAPI().getLanguages();
+				}
+			}
+		}
+		return languages;
+		
+	}
+	
+
+	   public static void makeBackendContext(Context context, IHTMLPage htmlPage, String cmsTemplateInode, String idURI, HttpServletRequest request,
+	           PageMode mode, Host host) throws DotDataException {
+	
+	       context.put("context", context);
+
+	        // stick some useful variables in the context
+	        if (htmlPage != null) {
+	            context.put("HTMLPAGE_INODE", String.valueOf(htmlPage.getInode()));
+	            context.put("HTMLPAGE_IDENTIFIER", String.valueOf(htmlPage.getIdentifier()));
+	            context.put("HTMLPAGE_TITLE", htmlPage.getTitle());
+	            context.put("HTMLPAGE_META", htmlPage.getMetadata());
+	            //http://jira.dotmarketing.net/browse/DOTCMS-6427
+	            context.put("HTMLPAGE_DESCRIPTION", htmlPage.getSeoDescription());
+	            context.put("HTMLPAGE_KEYWORDS", htmlPage.getSeoKeywords());
+	            context.put("HTMLPAGE_SECURE", String.valueOf(htmlPage.isHttpsRequired()));
+	            context.put("HTMLPAGE_REDIRECT", htmlPage.getRedirect());
+	            context.put("friendlyName", htmlPage.getFriendlyName());
+	            context.put("pageTitle", htmlPage.getTitle());
+                context.put("dotPageMode", mode);
+	            Date moddate = htmlPage.getModDate();
+
+	            moddate = new Date(moddate.getTime());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use the Java 8 Date and Time API instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2143)